pkg/karmadactl/apply.go

```diff
@@ -0,0 +1,189 @@
+			// factory must be initialized before the command runs
+			restConfig, err := karmadaConfig.GetRestConfig(o.KarmadaContext, o.KubeConfig)
+			if err != nil {
+				return err
+			}
+			kubeConfigFlags.WrapConfigFn = func(config *restclient.Config) *restclient.Config { return restConfig }
+			flags.Factory = kcmdutil.NewFactory(kubeConfigFlags)
+			return nil
+		},
+		Run: func(cmd *cobra.Command, args []string) {
+			// coverts flags to the kubectl apply options
+			o, err := flags.ToOptions(cmd, parentCommand, args)
+			kcmdutil.CheckErr(err)
+			// generate and inject policies
+			o.PreProcessorFn = func() error {
+				return generateAndInjectPolices(flags.Factory, o)
+			}
+			kcmdutil.CheckErr(o.Validate(cmd, args))
+			kcmdutil.CheckErr(o.Run())
+		},
+	}
+	flags.AddFlags(cmd)
+	cmd.Flags().StringVar(&o.KubeConfig, "kubeconfig", "", "Path to the control plane kubeconfig file.")
+	cmd.Flags().StringVar(&o.KarmadaContext, "karmada-context", "", "Name of the cluster context in control plane kubeconfig file.")
+	// Fixme(carlory): kubectl uses kubeConfigFlags as global command-line options (applies to all commands).
+	cmd.Flags().StringVarP(kubeConfigFlags.Namespace, flagNamespace, "n", *kubeConfigFlags.Namespace, "If present, the namespace scope for this CLI request")
+	return cmd
+}
+
+// generateAndInjectPolices generates and injects policies to the given resources.
+// It returns an error if any of the policies cannot be generated.
+func generateAndInjectPolices(f kcmdutil.Factory, o *apply.ApplyOptions) error {
+	// load the resources
+	infos, err := o.GetObjects()
+	if err != nil {
+		return err
+	}
+
+	// generate policies and append them to the resources
+	var results []*resource.Info
+	for _, info := range infos {
+		results = append(results, info)
+		obj := generatePropagationObject(info)
+		gvk := obj.GetObjectKind().GroupVersionKind()
+		mapping, err := o.Mapper.RESTMapping(gvk.GroupKind(), gvk.Version)
+		if err != nil {
+			return fmt.Errorf("unable to recognize resource: %v", err)
+		}
+		client, err := f.ClientForMapping(mapping)
+		if err != nil {
+			return fmt.Errorf("unable to connect to a server to handle %q: %v", mapping.Resource, err)
+		}
+		policyName, _ := metadataAccessor.Name(obj)
+		ret := &resource.Info{
+			Namespace: info.Namespace,
+			Name:      policyName,
+			Object:    obj,
+			Mapping:   mapping,
+			Client:    client,
+		}
+		results = append(results, ret)
+	}
+
+	// store the results object to be sequentially applied
+	o.SetObjects(results)
+	return nil
+}
+
+// generatePropagationObject generates a propagation object for the given resource info.
+// It takes the resource name and kind as the generated object name.
+// TODO(carlory): allow users to select clusters to propagate resources. default to all clusters.
+func generatePropagationObject(info *resource.Info) runtime.Object {
+	policyName := fmt.Sprintf("%s-%s", info.Name, strings.ToLower(info.Mapping.GroupVersionKind.Kind))
```

Comment:
A little suggestion,can you add a special prefix? such as `auto-`

Reply:
we can refer here https://github.com/karmada-io/karmada/pull/1824#discussion_r880012655

Reply:
@lonelyCZ @chaunceyjiang 

When generating a policy name, it's same as the `karmadactl promote` command.
If we want to indicate the policy name is generated by karmadactl, annotation may be a better choice. i.e. 

`karmadactl.karmada.io/generated-for-resource: "deployments.v1.apps/micro-dao-2048"` or `karmadactl.karmada.io/generated: ""`.

Reply:
If annotations is chosen, I'll fill another pull request for `apply` and `promote` commands.

@chaunceyjiang @lonelyCZ how do you think?

Reply:
> When generating a policy name, it's same as the karmadactl promote command.
If we want to indicate the policy name is generated by karmadactl, annotation may be a better choic

IMO, this is a good idea

I tend to the  `karmadactl.karmada.io/generated-for-resource`